code/game/objects/structures/morgue.dm

```diff
@@ -14,12 +14,17 @@
 	var/morgue_open = 0
 	anchored = 1
 	throwpass = 1
+	var/buildstacktype = /obj/item/stack/sheet/metal
+	var/buildstackamount = 2
 
 /obj/structure/morgue/Initialize()
 	. = ..()
 	connected = new tray_path(src)
 
 /obj/structure/morgue/Destroy()
+	new buildstacktype(loc, buildstackamount)
+	for(var/atom/movable/A in src)
```

Comment:
A needs a more descriptive variable name. Also please use contents instead of `src`.
The destroy proc should not have any side-effects or leave any lingering objects.. Separate this out into a different proc

Reply:
I have no idea how to move the buildstacktype beyond the Destroy() function. If the exact same line of code (with variables stacktype and stackamount declared) is added to the attack_alien proc just prior to qdel(src), it does not drop anything. Been trying to troubleshoot this for several hours.

Reply:
I thought the loc might be at fault but get_turf(src) doesn't address the issue either. I've tried adding a destroy() proc to the morgue (as in obj/structure/morgue/dstroy()) but I'm unsure how to directly reference that in attack_alien.dm as opposed to obj/structure/destroy() (i.e. the generic inherited proc).

Reply:
It's a problem in CM in general. There's no 'deconstruct' proc.

Reply:
Actually no, the deconstruct proc exists in the form of a lowercase 'destroy'
Someone should really change that name